app/controllers/users_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UsersController < ApplicationController
+  skip_before_action :ensure_login, only: :new
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.authenticate(params[:user][:password_confirmation]) && @user.save
```

Comment:
You don't need to authenticate here. has_secure_password will automatically check that password_confirmation matches password if you pass both into "new"